### PR TITLE
Update CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by harinikmsft**
Updating the contribution guide to be specific to MacOS contributions and outlining processes for opening up to welcome PRs from the community.

###### Microsoft Reviewers: [Open in CodeFlow](http://wpcp.azurewebsites.net/CodeFlowProtocolProxy2.php?pullrequest=https://github.com/microsoft/react-native/pull/325)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -1,108 +1,130 @@
...
-As a reminder, all contributors are expected to adhere to the [Code of Conduct](https://code.facebook.com/codeofconduct).
+- Navigate to **[GitHub](https://www.github.com)** with a browser and log in to your GitHub account. For the sake of this document, let's assume your username is **johndoe**.
+- Navigate to the **[microsoft/react-native](https://github.com/microsoft/react-native)** repository in the same browser session.
+- Click on the **Fork** button at the top right corner of the page.
+- Create the fork under your account. Your GitHub profile should now show **react-native** as one of your repositories.
+- Create a folder on your device and clone your fork of the **Microsoft** repository. e.g. `https://github.com/johndoe/react-native.git`. Notice how your GitHub username is in the repository location.
 
-## Ways to Contribute
+```
+> git clone https://github.com/johndoe/react-native.git
```
Comment (tom-un) [inline CONTRIBUTING.md:21]: Suggestion: drop the `> ` prefix before command line commands which is Windows-like and make the code block be ` ```bash ` which is the convention in facebook's setting up docs.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -1,108 +1,130 @@
...
+> git push
+```
+## Contributing
 
-The React Native blog is generated [from the Markdown sources for the blog](https://github.com/facebook/react-native-website/tree/master/website/blog).
+### Make the fix
+Now that your branch is set up and ready for commits, go ahead and fix the bug you are working on or make some small change that you want to check in. 
+ 
+### Verify your changes
+Manually test your fix, and ensure that all tests are in a good state
```
Comment (tom-un) [inline CONTRIBUTING.md:94]: Should we describe running the local automated test that will be run in CI when they make a pull request?
i.e.
yarn test  # run jest tests on JavaScript
yarn lint # run eslint on JavaScript
yarn flow-check-macos # run Flow checks on JavaScript

Manually test by running RNTester
Run Unit Tests and Integration Tests in the RNTesterPods Xcode project.

Comment (tom-un) [reply CONTRIBUTING.md:94]: Maybe TMI as if they will be caught by CI during the PR

Comment (harinikmsft) [reply CONTRIBUTING.md:94]: It is good to add these as best practices with a call out that it will be caught in CI. I will add.

**Comment by Saadnajmi** [comment]
The CONTRIBUTING.md file has been updated, but the top level of README.md has a large "contributing" link that redirects to facebook. Should it not be redirecting there?

**Comment by harinikmsft** [comment]
> The CONTRIBUTING.md file has been updated, but the top level of README.md has a large "contributing" link that redirects to facebook. Should it not be redirecting there?

There is a PR in progress to update the readme as well - [PR](https://github.com/microsoft/react-native-macos/pull/343)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -1,108 +1,136 @@
...
 
-## GitHub Issues
+**SourceTree:**
+If you are using SourceTree you will want your branch to show up in SourceTree so you can commit changes to your branch. It takes time for it to show up automatically, so you can make it show by running `git remote update origin –prune` in your command prompt from the root. Once you see your new branch in SourceTree under Remotes on the left navigation pane, double click on your branch to check it out locally. A dialog will come up and the default settings should be fine, click Ok.  
 
-We use GitHub issues to track bugs exclusively. We have documented our issue handling processes in the [Issues wiki](https://github.com/facebook/react-native/wiki/Issues).
+**Git Command Line**
+If you are using the command line, you will want to make sure you have your branch locally. It takes time for it to show up automatically, so you can make it show by running `git remote update origin –prune` in your command prompt from the root. Run `git branch -a` to see if your new branch shows up. Now you will want to check out your branch locally. You can do this with `git checkout -b branch-name`. Confirm you are now working out of the branch with `git branch`.
```
Comment (Saadnajmi) [inline CONTRIBUTING.md:65]: `git remote update origin –prune` is failing for me. Instead, I did `git pull --all`

Comment (harinikmsft) [reply CONTRIBUTING.md]: Updated, thanks.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -1,108 +1,136 @@
-# Contributing to React Native
+# Contributing to microsoft/react-native-macos
 
-Thank you for your interest in contributing to React Native! From commenting on and triaging issues, to reviewing and sending Pull Requests, all contributions are welcome. We aim to build a vibrant and inclusive [ecosystem of partners, core contributors, and community](ECOSYSTEM.md) that goes beyond the main React Native GitHub repository.
+This document describes how to set up your development environment and contribute changes to the **microsoft/react-native-macos** project. This is a working fork of **facebook/react-native** where changes for supporting MacOS are being staged.
```
Comment (tom-un) [inline CONTRIBUTING.md:3]: MacOS => macOS

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -1,108 +1,136 @@
-# Contributing to React Native
+# Contributing to microsoft/react-native-macos
 
-Thank you for your interest in contributing to React Native! From commenting on and triaging issues, to reviewing and sending Pull Requests, all contributions are welcome. We aim to build a vibrant and inclusive [ecosystem of partners, core contributors, and community](ECOSYSTEM.md) that goes beyond the main React Native GitHub repository.
+This document describes how to set up your development environment and contribute changes to the **microsoft/react-native-macos** project. This is a working fork of **facebook/react-native** where changes for supporting MacOS are being staged. 
+> **Note: This repository will be accepting PRs only specific to MacOS support. To contribute to React Native, please see [Contributing to react-native](https://github.com/facebook/react-native/blob/master/CONTRIBUTING.md)**
```
Comment (tom-un) [inline CONTRIBUTING.md:4]: macOS

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -1,108 +1,136 @@
...
 
-We also have a thriving community of contributors who would be happy to help you get set up. You can reach out to us through [@ReactNative](http://twitter.com/reactnative) (the React Native team) and [@ReactNativeComm](http://twitter.com/reactnativecomm) (the React Native Community organization).
+- Now running `git remote -v` should show the upstream repository also
 
-### Repositories
+```bash
+git remote -v
 
-The main repository, <https://github.com/facebook/react-native>, contains the React Native framework itself and it is here where we keep track of bug reports and manage pull requests.
+     origin  https://github.com/johndoe/react-native-macos.git (fetch)
```
Comment (tom-un) [inline CONTRIBUTING.md:49]: nit: tabs after the aliases are aligning the urls in a weird way.   Could just use spaces to manually align the urls on the same column.

Comment (tom-un) [reply CONTRIBUTING.md:49]: Never mind: it take it back.   The same alignment is exactly what one sees on the command line.  🤪

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -1,108 +1,136 @@
...
-* **Discussions** about the future of React Native take place in the <https://github.com/react-native-community/discussions-and-proposals> repository.
-* **High-quality plugins** for React Native can be found throughout the [React Native Community GitHub Organization](http://github.com/react-native-community/).
+### Setting up the branch
 
-Browsing through these repositories should provide some insight into how the React Native open source project is managed.
+For each bug or task you complete, it is recommended that you start with a fresh branch. If you have any lingering changes in your current branch that you want to save, go ahead and commit them. If you are just beginning, then you are good to go. On github, navigate to your repository which should be forked from microsoft/react-native-macos as described in the above sections. Above the list of files is a dropdown that should say master. Use the dropdown to create a new branch and name is according to what you will be working on. (I.e. DropdownHighlight, CleanUpExamples, etc). Now you have created a new branch.
```
Comment (tom-un) [inline CONTRIBUTING.md:59]: Make bold: **microsoft/react-native-macos** to be consistent with other instances.

Comment (harinikmsft) [reply CONTRIBUTING.md:59]: Done.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -1,108 +1,136 @@
...
-2. **Reviewing pull requests for the docs.** Reviewing [documentation updates](https://github.com/facebook/react-native-website/pulls) can be as simple as checking for spelling and grammar. If you encounter situations that can be explained better in the docs, click **Edit** at the top of most docs pages to get started with your own contribution.
-3. **Help people write test plans.** Some pull requests sent to the main repository may lack a proper test plan. These help reviewers understand how the change was tested, and can speed up the time it takes for a contribution to be accepted.
+- When you run `git remote -v`, you should see only your fork in the output list
 
-Each of these tasks is highly impactful, and maintainers will greatly appreciate your help.
+```bash
+git remote -v
 
-### Our Development Process
+     origin  https://github.com/johndoe/react-native-macos.git (fetch)
```
[the remark this answers is not in the document]
Comment (harinikmsft) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 5 above)
Comment (harinikmsft) [reply CONTRIBUTING.md]: Done

File: CONTRIBUTING.md (hunk 4 above)
Comment (harinikmsft) [reply CONTRIBUTING.md]: Done

**Comment by harinikmsft** [comment]
> Looks so good! Just made a few minor suggestions.

Cool. I think everything is addressed now.

**Pull request merged**